### ISSUE TITLE
Fix setAnalogControlState

### ISSUE THREAD

**Issue by LosFaul** [multitheftauto/mtasa-blue#567]
Edit: i cleaned up this topic, because it was a mess before

**Is your feature request related to a problem? Please describe.**
setAnalogControlState is not working when there is actual player input.

```lua
addEventHandler("onClientPreRender", root,
    function()
        local accelerate = getAnalogControlState("accelerate")
        setAnalogControlState("accelerate", accelerate * 0.1)
    end
)
```
in this example i try to modify the player input,
but setAnalogControlState has no effect

**Describe the solution you'd like**
Fix setAnalogControlState, so that it allows to modify player input

**Comment by LosFaul** [comment]
Today i made some testing again, to ensure how things work
(getAnalogControlState/setAnalogControlState)

`addEventHandler("onClientPreRender", root,
	function()
		local accelerate = getAnalogControlState("accelerate")
		setAnalogControlState("accelerate", accelerate * 0.1)
	end,
	false
)`

the new set value by setAnalogControlState is always overwritten and the use of this function has no effect
i'm not sure if it is inteded to work this way
but then there should be an additional argument to setAnalogControlState to not allow an additional overwrite for the current frame from input:
setAnalogControlState(controlName, value[, disableOverwriteForCurrentFrame = false])

**Comment by LosFaul** [comment]
this is a bump with another example:
reversing controls

```lua
addEventHandler("onClientPreRender", root,
    function()
        -- get controll values
        local left = getAnalogControlState("vehicle_left")
        local right = getAnalogControlState("vehicle_right")
        -- reverse controll values
        setAnalogControlState("vehicle_left", right)
        setAnalogControlState("vehicle_right", left)
    end
)
```
this does not work either

you can it get working for keyboards but **not for gamepads** as gamepads will bypass it
keyboard solution would be currently to unbind controll keys and make and own check if the keys are pressed

but if the issue would be fixed keyboard and gamepads would share the same solution

**Comment by qaisjp** [comment]
Remind me to self-assign 1 month from now. Issue is open to someone else until then.

**Comment by LosFaul** [comment]
I thought a bit how a potential solution could be done.
Keeping the game process order in mind: https://wiki.multitheftauto.com/wiki/Game_Processing_Order
and not trying to break existing scripts

a solution could look like this:
```lua
setAnalogControlState(string controll, float value[, forced = false])
```
the solution would be to add a third optional argument which is per default false
this is in order to not break existing scripts
if forced is true
the value is ensured to be used for the next game position update
and after the position update it should work normally again
so that you can get after the game position update in onClientPreRender event the player input and be able to transform it again

**Comment by qaisjp** [comment]
The API you propose sounds good to me.

**Comment by LosFaul** [comment]
i figured out player input update happens between onClientHUDRender and onClientRender
which is actually a bit questionable placement for player input update

but the issue still remains you can't transform player input with scripting

**Comment by ghost** [comment]
Fixed in #1852
